src/lms7002m/LMS7002M.cpp

```diff
@@ -1512,9 +1515,29 @@ int LMS7002M::SetFrequencySX(bool tx, float_type freq_Hz, SX_details* output)
         output->div_loch = div_loch;
     }
 
-    //find which VCO supports required frequency
+    // turn on VCO and comparator
     Modify_SPI_Reg_bits(LMS7param(PD_VCO), 0); //
-    Modify_SPI_Reg_bits(LMS7param(PD_VCO_COMP), 0); //
+    Modify_SPI_Reg_bits(LMS7param(PD_VCO_COMP), 0);
+
+    // try setting tuning values from the cache, if it fails perform full tuning
+    if  (tuning_cache_sel_vco.count(freq_Hz) > 0)
```

Comment:
I would prefer if caching would be optional, there is `useCache` flag. Same for storing the results.